driver/logcumsumexp_driver.hpp

```diff
@@ -0,0 +1,453 @@
+    inflags.AddInputFlag(
+        "dim", 'd', "0", "The dimension to do the operation over (Default=0)", "int");
+    inflags.AddInputFlag("exclusive",
+                         'e',
+                         "0",
+                         "Enable exclusive calculation. 0 for False, 1 for True (Default=0)",
+                         "int");
+    inflags.AddInputFlag(
+        "reverse",
+        'r',
+        "0",
+        "Reverse the calculation order to back to front. 0 for False, 1 for True (Default=0)",
+        "int");
+    inflags.AddInputFlag("Contiguous",
+                         'C',
+                         "1",
+                         "Is input tensor contiguous? (Default=1 for contiguous tensor)",
+                         "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "0", "Verify Each Layer (Default=0)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int LogCumSumExpDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t input_sz   = miopen::deref(inputDesc).GetElementSpace();
+    size_t output_sz  = miopen::deref(outputDesc).GetElementSpace();
+    size_t doutput_sz = miopen::deref(doutputDesc).GetElementSpace();
+    size_t dinput_sz  = miopen::deref(dinputDesc).GetElementSpace();
+
+    uint32_t ctx = 0;
+
+    input_dev   = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    output_dev  = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+    doutput_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, doutput_sz, sizeof(Tgpu)));
+    dinput_dev  = std::unique_ptr<GPUMem>(new GPUMem(ctx, dinput_sz, sizeof(Tgpu)));
+
+    input   = std::vector<Tgpu>(input_sz);
+    output  = std::vector<Tgpu>(output_sz);
+    doutput = std::vector<Tgpu>(doutput_sz);
+    dinput  = std::vector<Tgpu>(dinput_sz);
+
+    output_host = std::vector<Tref>(output_sz);
+    dinput_host = std::vector<Tref>(output_sz);
+
+    for(int i = 0; i < input_sz; i++)
+        input[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-1), static_cast<Tgpu>(1));
+
+    for(int i = 0; i < doutput_sz; i++)
+        doutput[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-1), static_cast<Tgpu>(1));
+
+    if(input_dev->ToGPU(GetStream(), input.data()) != 0)
+    {
+        std::cerr << "Error copying (input) to GPU, size: " << input_dev->GetSize() << std::endl;
+        return miopenStatusAllocFailed;
+    }
+
+    if(doutput_dev->ToGPU(GetStream(), doutput.data()) != 0)
+    {
+        std::cerr << "Error copying (doutput) to GPU, size: " << doutput_dev->GetSize()
+                  << std::endl;
+        return miopenStatusAllocFailed;
```

Comment:
ditto

Reply:
I've changed the return code
```suggestion
        return miopenStatusInternalError;
```